test/Microsoft.ComponentDetection.Detectors.Tests/PnpmDetectorTests.cs

```diff
@@ -783,6 +783,75 @@ public async Task TestPnpmDetector_V9_GoodLockVersion_SkipsFileAndLinkDependenci
             parentComponent => parentComponent.Name == "sampleDependency");
     }
 
+    [TestMethod]
+    public async Task TestPnpmDetector_V9_GoodLockVersion_FileAndLinkDependenciesAreNotRegistered()
+    {
+        var yamlFile = @"
+lockfileVersion: '9.0'
```

Comment:
@RushabhBhansali discovered exceptions for versions that look like:
- 'https://codeload.github.com/types/mocha/tar.gz/16bf4127041c71995b5662232b228842ce76ba82'
- 'https://cdn.sheetjs.com/xlsx-0.20.2/xlsx-0.20.2.tgz'
Are this type of parsing issue covered in this PR?